suite/tests/api/ir_riscv64.c

```diff
@@ -306,97 +306,120 @@ test_atomic(void *dc)
+                              opnd_create_reg(DR_REG_X31),
                               opnd_create_immed_int(0b11, OPSZ_2b));
     ASSERT(instr_is_exclusive_store(instr));
     test_instr_encoding(dc, OP_sc_d, instr);
 
     /* AMO */
     instr = INSTR_CREATE_amoswap_w(
-        dc, opnd_create_reg(DR_REG_X0), opnd_create_reg(DR_REG_X1),
+        dc, opnd_create_reg(DR_REG_X0),
+        opnd_create_base_disp(DR_REG_X1, DR_REG_NULL, 0, 0, OPSZ_4),
         opnd_create_reg(DR_REG_X31), opnd_create_immed_int(0b00, OPSZ_2b));
     test_instr_encoding(dc, OP_amoswap_w, instr);
-    instr = INSTR_CREATE_amoswap_d(dc, opnd_create_reg(DR_REG_X31),
-                                   opnd_create_reg(DR_REG_X1), opnd_create_reg(DR_REG_X0),
-                                   opnd_create_immed_int(0b01, OPSZ_2b));
+    instr = INSTR_CREATE_amoswap_d(
+        dc, opnd_create_reg(DR_REG_X31),
+        opnd_create_base_disp(DR_REG_X1, DR_REG_NULL, 0, 0, OPSZ_8),
+        opnd_create_reg(DR_REG_X0), opnd_create_immed_int(0b01, OPSZ_2b));
     test_instr_encoding(dc, OP_amoswap_d, instr);
-    instr = INSTR_CREATE_amoadd_w(dc, opnd_create_reg(DR_REG_X0),
-                                  opnd_create_reg(DR_REG_X31), opnd_create_reg(DR_REG_A2),
-                                  opnd_create_immed_int(0b10, OPSZ_2b));
+    instr = INSTR_CREATE_amoadd_w(
+        dc, opnd_create_reg(DR_REG_X0),
+        opnd_create_base_disp(DR_REG_X31, DR_REG_NULL, 0, 0, OPSZ_4),
+        opnd_create_reg(DR_REG_A2), opnd_create_immed_int(0b10, OPSZ_2b));
     test_instr_encoding(dc, OP_amoadd_w, instr);
-    instr = INSTR_CREATE_amoadd_d(dc, opnd_create_reg(DR_REG_A0),
-                                  opnd_create_reg(DR_REG_A1), opnd_create_reg(DR_REG_A2),
-                                  opnd_create_immed_int(0b11, OPSZ_2b));
+    instr = INSTR_CREATE_amoadd_d(
+        dc, opnd_create_reg(DR_REG_A0),
+        opnd_create_base_disp(DR_REG_A1, DR_REG_NULL, 0, 0, OPSZ_8),
+        opnd_create_reg(DR_REG_A2), opnd_create_immed_int(0b11, OPSZ_2b));
     test_instr_encoding(dc, OP_amoadd_d, instr);
-    instr = INSTR_CREATE_amoxor_w(dc, opnd_create_reg(DR_REG_A0),
-                                  opnd_create_reg(DR_REG_A1), opnd_create_reg(DR_REG_A2),
-                                  opnd_create_immed_int(0b01, OPSZ_2b));
+    instr = INSTR_CREATE_amoxor_w(
+        dc, opnd_create_reg(DR_REG_A0),
+        opnd_create_base_disp(DR_REG_A1, DR_REG_NULL, 0, 0, OPSZ_4),
+        opnd_create_reg(DR_REG_A2), opnd_create_immed_int(0b01, OPSZ_2b));
     test_instr_encoding(dc, OP_amoxor_w, instr);
-    instr = INSTR_CREATE_amoxor_d(dc, opnd_create_reg(DR_REG_A0),
-                                  opnd_create_reg(DR_REG_A1), opnd_create_reg(DR_REG_A2),
-                                  opnd_create_immed_int(0b01, OPSZ_2b));
+    instr = INSTR_CREATE_amoxor_d(
+        dc, opnd_create_reg(DR_REG_A0),
+        opnd_create_base_disp(DR_REG_A1, DR_REG_NULL, 0, 0, OPSZ_8),
```

Comment:
Are usages with non-zero index or displacement supported? Add tests if yes.

Reply:
No, they're not supported, amo instructions only have a base register.